### PR TITLE
atom features function and helper functions for DMPNN Featurizer

### PR DESCRIPTION
**Pull request by ARY2260**
# Pull Request Template

## Description

This PR contains code for the atom_features() function and other required helper functions which will be helpful in making the DMPNN featurizer. It also contains a file with unit tests for all the added functions. 

This will be the first PR for GSOC'22 Project: D-MPNN Model Implementation

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/molecule_featurizers/dmpnn_featurizer.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  -------
+  features: List[Union[bool, int, float]]
+    An list of atom features.
+
+  Examples
+  --------
+  >>> from rdkit import Chem
+  >>> mol = Chem.MolFromSmiles('C')
+  >>> atom = mol.GetAtoms()[0]
+  >>> features = dc.feat.molecule_featurizers.dmpnn_featurizer.atom_features(atom)
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/dmpnn_featurizer.py:128]: You need to create an actual DeepChem featurizer object that inherits from `MolecularFeaturizer`. See for example https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#deepchem.feat.ConvMolFeaturizer. Are you planning to do this in a future PR?

Comment (rbharath) [reply deepchem/feat/molecule_featurizers/dmpnn_featurizer.py:128]: Ok reading your comment I think you are planning to do this in a future PR if I understand correctly

Comment (ARY2260) [reply deepchem/feat/molecule_featurizers/dmpnn_featurizer.py:128]: I plan to implement featurizer support functions, and graph object class for this week and will try to implement the DMPNNFeaturizer class and DMPNNEncoding class next week.

**Comment by ARY2260** [comment]
None of the errors in checks seemed to be directly linked to the code modified. One of the errors that popped up multiple times is related to TensorBoard import error

**Comment by arunppsg** [comment]
> None of the errors in checks seemed to be directly linked to the code modified. One of the errors that popped up multiple times is related to TensorBoard import error

The CI errors on tensorboard import have been fixed in 7419599. Can you do a git rebase and push to update your branch with the new changes in master now? (On DMPNN_featurizer branch in local, `git fetch upstream; git rebase upstream/master; git push upstream DMPNN_featurizer`)

**Comment by ARY2260** [comment]
> > None of the errors in checks seemed to be directly linked to the code modified. One of the errors that popped up multiple times is related to TensorBoard import error
> 
> The CI errors on tensorboard import have been fixed in [7419599](https://github.com/deepchem/deepchem/commit/741959977da2024fa8c0cdd2bab00b14f1055ead). Can you do a git rebase and push to update your branch with the new changes in master now? (On DMPNN_featurizer branch in local, `git fetch upstream; git rebase upstream/master; git push upstream DMPNN_featurizer`)

I had to force to my remote branch (git push -f origin DMPNN_featurizer) to apply the necessary changes. WIll this be fine?

**Comment by ARY2260** [comment]
@JoseAntonioSiguenza and @shivance, this PR contains atom features function and helper functions and suitable tests for the same. I have also added the docstrings and type annotations.

**Comment by arunppsg** [comment]
> I had to force push to my remote branch (git push -f origin DMPNN_featurizer) to apply the necessary changes. WIll this be fine? If yes, could you please run the checks again.

That's fine.

**Comment by ARY2260** [comment]
I pulled the latest changes from the upstream by mistake and tried to undo that. In this process, I lost the result of last workflow run on this PR. Last time : 30 tests were passed, 3 failed, and 2 were skipped.

**Comment by rbharath** [comment]
Looks like failures are unrelated to good to merge in!

**Pull request merged**